client/app/components/edit-service-modal/edit-service-modal-service.factory.js

```diff
@@ -36,34 +37,48 @@
   function EditServiceModalController(serviceDetails, $state, $modalInstance, CollectionsApi, EventNotifications) {
     var vm = this;
 
-    vm.service = serviceDetails;
+    vm.isPristine = isPristine;
```

Comment:
Could you elaborate on this idiom? I don't see the advantage of using `extend` here as opposed to setting these properties directly on the controller's scope.